### ISSUE TITLE
Audio stops when context switching to Edisyn panel

### ISSUE THREAD

**Issue by ametersky** [eclab/edisyn#38]
On Ableton Live 11 (OS X 10.13.6) if i have a track running and it is outputting audio and my focused window is Ableton when I `CMD + Tab` to an Edisyn panel or click the Edisyn icon, the audio output of the track from Ableton stops.  It does not pause the Ableton track, but the audio does dissapear.  Restarting the track lets the audio return.

**Comment by eclab** [comment]
I so happen to be running almost exactly the same configuration: Ableton 10 on 10.13.6.  I can't seem to reproduce this.  Haven't installed 11 yet.  This is a pretty strange bug, because Edisyn is bog-standard Java MIDI plus CoreMIDI4Java, and has no audio code at all.

- What MIDI device is Edisyn sending to?  Receiving from?  What happens if you Disconnect MIDI in Edisyn? [See the MIDI menu]/

- Is your track also outputting or set to receive MIDI?  Is it using the same device that Edisyn is?

- What audio devices is Ableton outputting to? Any chance your audio device is also a MIDI device, such a USB Audio interface?

**Comment by ametersky** [comment]
Here's some more info + slight change to what is occuring.  If i have two midi clips to play a note in succession and I switch to Edisyn in the middle of clip 1, the audio will cut out, but will resume when clip 2 starts.

<img width="151" alt="image" src="https://user-images.githubusercontent.com/13753810/128657181-070a123e-6035-46d8-9610-f16f05cb9338.png">

So in this image above, audio cuts halfway btwn clip 1 and restarts when clip 2 happens.

Here are the Edisyn Midi settings this is over USB
<img width="413" alt="image" src="https://user-images.githubusercontent.com/13753810/128657276-3165b5ae-be12-44af-8f3c-928f5edc1256.png">

If I disconnect the MIDI then audio continues as normal with context switch. No issues...

Here are the Ableton settings 

<img width="892" alt="image" src="https://user-images.githubusercontent.com/13753810/128657367-de9e1279-afcc-40d0-b830-04bb95002394.png">

Track gets MIDI in from a midi keyboard on ch 2 (when set to IN) and sends out to blofeld on channel 5 via USB

Ableton is outputting MIDI to a Profire2626 which is also the I/O for audio, but the Blofeld is not connected to this system except for sending audio to it.
----
Also, I can replicate this same behavior with a Microwave XT and this is with MIDI cables, not USB.  The microwave has a MIDI IN and OUT connected to the Profire2626

**Comment by eclab** [comment]
> Ableton is outputting MIDI to a Profire2626 which is also the I/O for audio, but the Blofeld is not connected to this system except for sending audio to it.

I'm confused by this statement.  In the pictures clearly Ableton is outputting MIDI to the Blofeld.  What happens if the Profire is entirely disconnected and Ableton is only talking to the Blofeld (and so is Edisyn)?  I can check that on my Blofeld tomorrow morning.

> Also, I can replicate this same behavior with a Microwave XT and this is with MIDI cables, not USB. The microwave has a MIDI IN and OUT connected to the Profire2626

I presume the Blofeld is not connected here, and so Edisyn is also trying to talk to the XT through the Profire?

What happens if you set up Ableton's tracks to only output to one device (say, the Profire), and Edisyn only to output and input from the Blofeld?  I'm guessing it'll work.

My guess is that Abelton 11 has a bug: if some other application grabs the MIDI device to output to it, Ableton pauses until it has sole control over the device again.

Alternatively, if Ableton and Edisyn play nicely when the Blofeld is the sole device, but badly whenever the Profire is running, this could be a Profire driver issue.

Sean

**Comment by ametersky** [comment]
> I'm confused by this statement.  In the pictures clearly Ableton is outputting MIDI to the Blofeld.  What happens if the Profire is entirely disconnected and Ableton is only talking to the Blofeld (and so is Edisyn)?  I can check that on my Blofeld tomorrow morning.

Sorry, I meant disconnected in the sense that all the MIDI is happening over USB and not through the Profire.

> I presume the Blofeld is not connected here, and so Edisyn is also trying to talk to the XT through the Profire?

Yes, Blofeld is not connected to the Profire and the Microwave is.

Fwiw,  I did not have my Blofeld connected earlier today and it was just the Microwave running MIDI through the Profire and it was having the same issue.

> What happens if you set up Ableton's tracks to only output to one device (say, the Profire), and Edisyn only to output and input from the Blofeld?  I'm guessing it'll work.

Just shut everything down for the night, but will try tomorrow and report back.

Thanks for your help.  I've been using Edisyn for quite some time, it's a really incredible piece of software :)

**Comment by eclab** [comment]
On Aug 8, 2021, at 11:59 PM, ametersky ***@***.***> wrote:
> I'm confused by this statement. In the pictures clearly Ableton is outputting MIDI to the Blofeld. What happens if the Profire is entirely disconnected and Ableton is only talking to the Blofeld (and so is Edisyn)? I can check that on my Blofeld tomorrow morning.
> 
> Sorry, I meant disconnected in the sense that all the MIDI is happening over USB and not through the Profire
> 
Okay but once again, what happens if the Profire is *entirely* disconnected? Not just "not used", but literally unplugged, audio is routed to the laptop speakers, and all MIDI is routed to the Bofeld?  Does it still fail?

Also: have you tried any other Blofeld editor, like (say) Bigglesworth, in this configuration?  Does it still fail?

Basically trying to determine if this is a Ableton 11 stupidity/bug or a Profire driver bug.  The chance that it's a Java/CoreMIDI4Java error seems remote.

Sean

**Comment by ametersky** [comment]
Okay, so I have the Profire2626 unplugged, only the Blofeld is connected via USB for MIDI I/O and the Blofeld audio out going to some headphones.  I don't think there's another way to hear the audio without the interface bc the Blofeld doesn't transmit audio over USB.

Still seeing the same issue.  When playing a MIDI clip that sustains a note when I switch from Ableton to the Edisyn panel, the audio cuts.  

Sounds like an Ableton bug based on what you said earlier?

**Comment by eclab** [comment]
Ohhhhhh, this is when you have a MIDI note that's *sustained*!   As opposed to an *Ableton audio stream* that's being cut in some way.  I know exactly what this is.

As a safety measure, Edisyn sends an All Sounds Off to every channel whenever you bring one of its windows to the front.  This allows it to guarantee that any new information the panel is sending to your synth isn't confused with past data just sent elsewhere (probably by another Edisyn panel).  So what's happening here is that when you bring Edisyn to the front, it's cutting off the sustained MIDI note.

I can set up an option in Edisyn to turn off this behavior for people in  your situation.  You have to use it with care of course.  Send me email directly ***@***.***) and I'll get you a link to a beta with the feature as soon as I can get it compiled.

Sean

**Comment by ametersky** [comment]
ah! that would be amazing. your email came through as all asterisks in the github comment.  Will the one in the Edisyn readme be okay?

**Comment by eclab** [comment]
Yes, or on the front page of the manual.
